spec/datetimeformat.html

```diff
@@ -286,43 +273,517 @@ <h1>Internal slots</h1>
+            </tr>
+            <tr>
+              <td>[[long]]</td>
+              <td>a DateTime Format Record</td>
+              <td>Format record for the *"long"* style.</td>
+            </tr>
+            <tr>
+              <td>[[medium]]</td>
+              <td>a DateTime Format Record</td>
+              <td>Format record for the *"medium"* style.</td>
+            </tr>
+            <tr>
+              <td>[[short]]</td>
+              <td>a DateTime Format Record</td>
+              <td>Format record for the *"short"* style.</td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+
+      <emu-clause id="sec-datetimeformat-connector-record">
+        <h1>DateTime Connector Records</h1>
+
+        <p>
+          Each <dfn id="datetimeformat-connector-record">DateTime Connector Record</dfn> has the fields defined in <emu-xref href="#table-datetimeformat-connector-record"></emu-xref>. All connector pattern strings must contain the strings *"{0}"* and *"{1}"*.
+        </p>
+        <emu-table id="table-datetimeformat-connector-record">
+          <emu-caption>DateTime Connector Record</emu-caption>
+          <table class="real-table">
+            <thead>
+              <tr>
+                <th>Field Name</th>
+                <th>Value Type</th>
+                <th>Description</th>
+              </tr>
+            </thead>
+            <tr>
+              <td>[[full]]</td>
+              <td>a Pattern String</td>
+              <td>Connector pattern when the date style is *"full"*.</td>
+            </tr>
+            <tr>
+              <td>[[long]]</td>
+              <td>a Pattern String</td>
+              <td>Connector pattern when the date style is *"long"*.</td>
+            </tr>
+            <tr>
+              <td>[[medium]]</td>
+              <td>a Pattern String</td>
+              <td>Connector pattern when the date style is *"medium"*.</td>
+            </tr>
+            <tr>
+              <td>[[short]]</td>
+              <td>a Pattern String</td>
+              <td>Connector pattern when the date style is *"short"*.</td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+
+      <emu-clause id="sec-datetimeformat-date-range-record">
+        <h1>DateTime Date Range Records</h1>
+
+        <p>
+          Each <dfn id="datetimeformat-date-range-record">DateTime Date Range Record</dfn> has the fields defined in <emu-xref href="#table-datetimeformat-date-range-record"></emu-xref>.
+        </p>
+        <emu-table id="table-datetimeformat-date-range-record">
+          <emu-caption>DateTime Date Range Record</emu-caption>
+          <table class="real-table">
+            <thead>
+              <tr>
+                <th>Field Name</th>
+                <th>Value Type</th>
+                <th>Description</th>
+              </tr>
+            </thead>
+            <tr>
+              <td>[[full]]</td>
+              <td>a DateTime Time Range Record</td>
```

Comment:
I'm also thinking about flattening these from [[DateTimeRangeFormat]].[[\<dateStyle>]].[[\<timeStyle>]] to [[DateTimeRangeFormat]].[[\<dateAndTimeStyle>]] with e.g. `Let _dateAndTimeStyle_ be the string-concatenation of _dateStyle_, *"-"*, and _timeStyle_.`, but haven't yet decided if that's worthwhile.